torch/ao/pruning/_experimental/pruner/base_structured_sparsifier.py

```diff
@@ -56,18 +165,60 @@ def _prepare(self, *args, **kwargs) -> None:
             )
             self.state[config["tensor_fqn"]]["mask"] = mask
             parametrize.register_parametrization(
-                module, tensor_name, parametrization(mask), unsafe=True
+                module, tensor_name, parametrization(mask)
             )
-
             prune_bias = config.get("prune_bias", True)
-            if prune_bias and module.bias is not None:
+            if module.bias is not None:
                 module.register_parameter("_bias", nn.Parameter(module.bias.detach()))
                 module.bias = None
+                module.prune_bias = prune_bias
+
             self.bias_handles.append(
                 module.register_forward_hook(
                     BiasHook(module.parametrizations.weight[0], prune_bias)
                 )
             )
 
-    def convert(self):
-        pass
+    def prune(self):
+        r"""
+        This function will FX symbolically trace the model and then find instances of the patterns
+        defined in self.patterns (by default SUPPORTED_STRUCTURED_PRUNING_PATTERNS ).
+
+        For each pattern, it will apply to corresponding conversion function, which will modify the output
+        and input size expected by the modules within the pattern
+        """
+
+        self.traced = symbolic_trace(self.model)
+        modules = dict(self.traced.named_modules())
+
+        # Right now we check for matches simply by iterating across all the patterns
+        # if this is slow we can store patterns in a trie-structure and modify this code for faster lookup
+
+        for node in self.traced.graph.nodes:
+            for pattern, convert_fn in self.patterns.items():
+                matched = apply_match(modules, pattern, node, [])
+                if matched is None:
+                    continue
+
+                first_module = modules.get(node.target)
+                # check if first module exists and has apropriate parameterization, otherwise skip
+                if (
+                    first_module is not None
+                    and parametrize.is_parametrized(first_module)
+                    and isinstance(first_module.parametrizations["weight"][0], FakeStructuredSparsity)
+                ):
+                    convert_block = []
+                    for node in matched:
+                        if node.op == "call_module":
+                            convert_block.append(modules.get(node.target))
+                        elif node.op == "call_function":
+                            convert_block.append(node.target)
+                    convert_fn(*convert_block)
+
+        # remove bias hooks, since biases are propogated during module conversion.
```

Comment:
can this be done during module conversion instead?  Either in `convert_fn` or directly after it.